mwdb/core/search/fields.py

```diff
@@ -84,16 +84,30 @@ def get_condition(self, expression: Expression, remainder: List[str]) -> Any:
             raise FieldNotQueryableException(
                 f"Field doesn't have subfields: {'.'.join(remainder)}"
             )
+
+        units = {"B": 1, "KB": 1024, "MB": 1048576, "GB": 1073741824}
+
+        def parse_size(size):
```

Comment:
It would be better to:
1. Put `is_digit` here and return the size if it matches
2. `raise UnsupportedGrammarException` unless it is digit or regex matches.